File: settlement/room/spirit/grave/ROOM_GRAVEYARD.java
package settlement.room.spirit.grave;

import java.io.IOException;

import settlement.main.SETT;
import settlement.path.finders.SFinderRoomService;
import settlement.room.main.RoomBlueprintIns;
import settlement.room.main.category.RoomCategorySub;
import settlement.room.main.furnisher.Furnisher;
import settlement.room.main.util.RoomInitData;
import snake2d.util.file.FileGetter;
import snake2d.util.file.FilePutter;
import snake2d.util.sets.LISTE;
import view.sett.ui.room.UIRoomModule;

public final class ROOM_GRAVEYARD extends RoomBlueprintIns<GraveInstance> implements GraveData.GRAVE_DATA_HOLDER{

	private final GraveData data;
	final CGraveyard constructor;
	private final SFinderRoomService finder;
	
	public ROOM_GRAVEYARD(int typeIndex, String key, RoomInitData init, RoomCategorySub block, SFinderRoomService finder) throws IOException {
		super(typeIndex, init, key, block);
		data = new GraveData(this, init, 20) {
			
			@Override
			public double respect(GraveInstance grave) {
				return constructor.respekk.get(grave);
			}

		};

		
		
		constructor = new CGraveyard(this, init);
		
		this.finder = finder;
	}

	@Override
	protected void update(double ds) {
		data.update(ds);
	}

	@Override
	public Furnisher constructor() {
		return constructor;
	}
	
	@Override
	public SFinderRoomService service(int tx, int ty) {
		return finder;
	}

	@Override
	protected void saveP(FilePutter file){
		data.save(file);
	}
	
	@Override
	protected void loadP(FileGetter file) throws IOException{
		data.load(file);
	}
	
	@Override
	protected void clearP() {
		data.clear();
	}

	@Override
	public void appendView(LISTE<UIRoomModule> mm) {
		//mm.add(new Gui(this).make());
	}

	@Override
	public GraveData graveData() {
		return data;
	}
	
	public boolean isGraveHead(int tx, int ty) {
		return is(tx, ty) && SETT.ROOMS().fData.tileData.get(tx, ty) == Grave.DIG_MARK;
	}

}

File: LegacySource/settlement/room/spirit/grave/ROOM_GRAVEYARD.cs
using System;
using System.IO;
using System.Collections.Generic;

namespace Settlement.Room.Spirit.Grave
{
    public class ROOM_GRAVEYARD : RoomBlueprintIns<GraveInstance>, GraveData.GRAVE_DATA_HOLDER
    {
        private readonly GraveData data;
        private readonly CGraveyard constructor;
        private readonly SFinderRoomService finder;

        public ROOM_GRAVEYARD(int typeIndex, string key, RoomInitData init, RoomCategorySub block, SFinderRoomService finder) : base(typeIndex, init, key, block)
        {
            data = new GraveData(this, init, 20)
            {
                Respect = grave => constructor.Respekk[grave]
            };

            constructor = new CGraveyard(this, init);
            this.finder = finder;
        }

        protected override void Update(double ds)
        {
            data.Update(ds);
        }

        public override Furnisher Constructor()
        {
            return constructor;
        }

        public override SFinderRoomService Service(int tx, int ty)
        {
            return finder;
        }

        protected override void SaveP(FilePutter file)
        {
            data.Save(file);
        }

        protected override void LoadP(FileGetter file)
        {
            data.Load(file);
        }

        protected override void ClearP()
        {
            data.Clear();
        }

        public override void AppendView(LISTE<UIRoomModule> mm)
        {
            //mm.Add(new Gui(this).Make());
        }

        public override GraveData GraveData()
        {
            return data;
        }

        public bool IsGraveHead(int tx, int ty)
        {
            return Is(tx, ty) && SETT.ROOMS().FData.TileData.Get(tx, ty) == Grave.DIG_MARK;
        }
    }
}